app/controllers/api/trainees/withdraws_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Api
+  module Trainees
+    class WithdrawsController < Api::BaseController
+      def create
+        if trainee.blank?
+          render(json: { error: "Trainee not found" }, status: :not_found)
+        else
+          trainee.update(withdrawal_params) if withdraw_allowed?
```

Comment:
Should this endpoint return 422 if withdrawal is not allowed?

Reply:
Do we need to send a request to DQT? I wondered whether there was a service that already does the basic withdraw trainee logic and I didn't see one. There is a `Dqt::WithdrawTraineeJob` service and I guess we need to call that at some point?

Reply:
Added 422
Added the rest of the withdraw logic